test/server/api_listener_test.cc

```diff
@@ -163,21 +163,26 @@ name: test_api_listener
 
   ASSERT_EQ("test_api_listener", http_api_listener.name());
   ASSERT_EQ(ApiListener::Type::HttpApiListener, http_api_listener.type());
-  ASSERT_TRUE(http_api_listener.http().has_value());
+  auto api_listener = http_api_listener.createHttpApiListener(server_.dispatcher());
```

Comment:
Can we make the wrapper class public and remove the outer readCallbacksForTest with a dynamic cast to the wrapper class here?

Reply:
Done.